perllib/FixMyStreet/Cobrand/Smidsy.pm

```diff
@@ -416,6 +416,307 @@ sub reports_hook_restrict_bodies_list {
+            $c->stash->{problems_rs} = $rs->search( { extra => \$filter } );
+        }
+    }
+
+    if ( my $area = $c->get_param('area') ) {
+        $c->stash->{area} = $area;
+        my $rs = $c->stash->{problems_rs};
+        if (my $body = $c->get_param('area_body') and $area ne 'wards') {
+            $c->stash->{area_body} = $body;
+            $c->stash->{problems_rs} = $rs->search( areas => { 'like', '%,' . $area . ',%' } );
+            $c->stash->{children} = $self->get_body_children( $body );
+        } else {
+            $area = $c->get_param('area_body') if $area eq 'wards';
+            $c->stash->{area_body} = $area;
+            $c->stash->{problems_rs} = $rs->search( { bodies_str => $area } );
+            $c->stash->{children} = $self->get_body_children( $area );
+        }
+    }
+
+    if ( $c->get_param('csv') ) {
+        $c->stash->{csv} = {
+            problems => $c->stash->{problems_rs}->search_rs({}, {
+                order_by => { '-desc' => 'me.confirmed' },
+            }),
+            headers => [
+                'Report ID',
+                'Title',
+                'Category',
+                'Created',
+                'Confirmed',
+                'Status',
+                'Latitude', 'Longitude',
+                'Query',
+                'Report URL',
+                'Source',
+                'Participants',
+            ],
+            columns => [
+                'id',
+                'title',
+                'category',
+                'created',
+                'confirmed',
+                'state',
+                'latitude', 'longitude',
+                'postcode',
+                'url',
+                'external_body',
+                'participants'
+            ],
+            extra_data => sub {
+                my $report = shift;
+
+                my $data = $report->get_extra_metadata;
+
+                return {
+                    external_body => $report->external_body,
+                    participants => $data->{participants} || '',
+                };
+            },
+            filename => 'collideoscope-data',
+        };
+        $self->generate_csv;
+        return 1;
+    } else {
+        $c->stash->{template} = 'reports/index.html';
+        unless ( $c->stash->{start_date} && $c->stash->{end_date} ) {
+            $c->forward('/reports/load_dashboard_data');
+            $c->stash->{problems_reported} = $c->stash->{problems_reported_by_period}->[-1];
+        } else {
+
+            $start = DateTime::Format::Strptime->new(
+                pattern => '%F', # yyyy-mm-dd
+            )->parse_datetime($c->stash->{start_date});
+            my $loop_start = $start->clone; # to avoid loop_period changing start
+
+            $end = DateTime::Format::Strptime->new(
+                pattern => '%F', # yyyy-mm-dd
+            )->parse_datetime($c->stash->{end_date});
+
+            use FixMyStreet::Script::UpdateAllReports;
+
+            my ($period, $extra);
```

Comment:
Again, feels like this should be factored with the identical/similar code in UpdateAllReports.